backend/corpora/lambdas/api/v1/authentication.py

```diff
@@ -0,0 +1,82 @@
+from flask import make_response, jsonify, current_app, request, redirect
+import json
+from authlib.integrations.flask_client import OAuth
+from urllib.parse import urlencode
+from functools import lru_cache
+import base64
+from ....common.authorizer import get_userinfo, assert_authorized_token
+from ....common.corpora_config import CorporaAuthConfig
+
+
+@lru_cache(maxsize=1)
+def create_oauth_client(config):
+    oauth = OAuth(current_app)
+    api_base_url = config.api_base_url
+    client = oauth.register(
+        "oauth",
+        client_id=config.client_id,
+        client_secret=config.client_secret,
+        api_base_url=api_base_url,
+        access_token_url=f"{api_base_url}/oauth/token",
+        authorize_url=f"{api_base_url}/authorize",
+        client_kwargs={"scope": "openid profile email offline"},
+    )
+    return client
+
+
+def login():
+    """api call,  initiate the login process"""
+    config = CorporaAuthConfig()
+    client = create_oauth_client(config)
+    callbackurl = f"{config.callback_base_url}/v1/oauth2/callback"
+    response = client.authorize_redirect(redirect_uri=callbackurl)
+    return response
+
+
+def logout():
+    """api call,  logout of the system"""
+    config = CorporaAuthConfig()
+    client = create_oauth_client(config)
```

Comment:
Can this oauth_client be reused, or do we need to create a new one for each api call?

Reply:
There should only be one.  create_oauth_client returns one that has already been created.